File: app/dbmodels/Consumable.php
<?php

$table = array(
    'fields'        => array(),
    'oldfields'     => array(),
    'renamefields'  => array(),
    'indexes'       => array(),
    'oldindexes'    => array(),
    'relationships' => array(),
    'model'         => 'Consumable',
    'menu'          => ''
);
$table['fields'] = array(
    'id'                 => DBModels::type('increments',
                                           array('visible' => false)),
    'consumable_item_id'  => DBModels::type('integer'),
    'date_in'            => DBModels::type('date'),
    'date_out'           => DBModels::type('date'),
    'item_type'           => DBModels::type('string',
                                           array('visible' => false,
                                                 'size'    => 100)), /// User or Group ? Define both relations ?
    'item_id'            => DBModels::type('integer',
                                           array('visible' => false)),
);

$table['relationships'] = array(
    'consumable_item' => array(
        'type' => 'belongsTo',
        'item' => 'ConsumableItem'),
    'infocom' => array(
        'type' => 'morphMany',
        'item' => 'Infocom'),
    'receiver' => array(
        'type' => 'morphTo'),
);
include "_commonentitylink.php";
include "_commonrecursiveentitylink.php";

File: fusionglpi/database/consumable.py
from fusionglpi import db


class Consumable(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    entity_id = db.Column(db.Integer, db.ForeignKey('entity.id'))
    is_recursive = db.Column(db.Boolean, nullable=False, server_default="0")
    consumable_item_id = db.Column(db.Integer,
                                   db.ForeignKey('consumable_item.id'))
    date_in = db.Column(db.Date)
    date_out = db.Column(db.Date)
    item_type = db.Column(db.String(255, None, True))
    item_id = db.Column(db.Integer, nullable=False, server_default="0")
